Fofura

erickotsuka opened the pull request.

erickotsuka committed:
[#9] Add instructions for adding submodules

erickotsuka committed:
[#9] Remove submodule

erickotsuka committed:
[#9] Arroz doce

d-nery commented on README.md:
```diff
@@ -139,3 +139,14 @@ opções da lista para executar os comandos de compilação e gravação mais ra
 
 > Em breve
+
+## Adding a submodule
```
Põe antes do debug que é em breve hahaha

freitas-renato replied:
Não é melhor deixar em português igual o resto do readme?

d-nery replied:
Justo haahah

erickotsuka replied:
Eita. Nem percebi ~porque sou muito bilíngue~.

erickotsuka committed:
[#9] Change order in README

erickotsuka committed:
[#9] Add README instructions for project name

erickotsuka committed:
[#9] Translate README instruction to pt-BR

erickotsuka commented:
E aí? Merge?

erickotsuka commented:
Ops, não dá merge ainda rs

d-nery commented on Makefile:
```diff
@@ -284,11 +287,19 @@ clean_all:
-	$(AT)clang-format -i $(C_SOURCES) $(wildcard */*.h)
-	@echo "Done"
+# Uncrustify auxiliary variables
+UNCRUSTIFY_FILES = $(C_SOURCES) $(wildcard */*.h)
```
find, sei lá se aqui tem mais de um nível de pasta

d-nery commented on inc/mcu.h:
```diff
@@ -36,4 +36,4 @@ void mcu_sleep(uint32_t ms);
 void led_toggle(void);
 
-#endif  // __MCU_H__
+#endif /* __MCU_H__ */
```
Ah não, tira essa regra do uncrustify tbm hahaha é muito chata

d-nery commented on Makefile:
```diff
@@ -284,11 +287,19 @@ clean_all:
+UNCRUSTIFIED_SOURCES = $(UNCRUSTIFY_FILES:%=.uncrustify/%)
+
+# Uncrustify auxiliary target
+.uncrustify/%: %
```
Entendi nada mas acredito

erickotsuka replied:
Só explicando. O uncrustify cria um outro arquivo com as correções de formatação. Aí cria na pasta .uncrustify e depois copia de volta pro lugar certo. Depois, apaga a pasta uncrustify. E não dá pra passar uma lista de arquivos, é um arquivo por vez.

d-nery replied:
uashuash ok

erickotsuka committed:
[#9] Fix adding submodule example in README

erickotsuka committed:
[#9] Check existence of submodule dir before including

erickotsuka committed:
[#9] Change formatter to uncrustify

- Add uncrustify.cfg
- Remove .clang-format
- Update pre-commit hook
- Update Makefile format rule

d-nery commented:
Ou, tem que tirar o clang-format dos requisitos então, e adicionar o uncrustify

erickotsuka committed:
[#9] Change formatting tool in requirements

freitas-renato reviewed (approved):
Parece estar tudo certo.

erickotsuka commented:
Testei no Windows com MSYS2 e funcionou as coisas do uncrustify.
